### PR TITLE
[PROF-11035] Bump to version 3.3.7

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**

This PR bumps the gem to version 3.3.7 in preparation for the next release.

**Motivation:**

Release a new version with the Ruby 3.4.0-rc1 headers from #8.

**Additional Notes:**

I've also removed the `CHANGELOG.md` and `CONTRIBUTING.md` files as they were leftovers from our forking `debase-ruby_core_source` and aren't particularly relevant anymore -- we have our own separate changelog file, and the instructions in `CONTRIBUTING.md` are also in the `README.md`.

This PR is on top of #8 to make it easier to test the new release (and because it implicitly depends on #8 anyway); once #8 gets merged to master, this one will automatically will get changed to master as well.

**How to test the change?**

Here's what I did to validate this locally:

1. Ran `bundle exec rake build` to package the gem
2. Manually installed the packaged gem, and ran the profiler test suite on ruby 3.4.0-rc1 and 3.3.5 with this version.

### CODE REVIEW

**Review by anmarchenko — APPROVED** [review]

**Comment by ivoanjo** [comment]
We should only merge this and release once #11 is merged as well.

**Review by AlexJF — APPROVED** [review]

**Pull request merged**
